docs(api): Python Protocol API 2.18

ecormany opened the pull request:
# Overview

This is the feature branch for Python API version 2.18 docs.

_Please base all PRs with 2.18-related docs here, rather than on `edge`, if possible._ (But get code into whatever branch makes sense.)

ecormany committed:
docs(api): section on Position Relative to Trash Containers (#14593)

# Overview

Follow up to further explain behavior introduced in #14560.

Addresses RTC-400.

# Test Plan

-
[sandbox](http://sandbox.docs.opentrons.com/docs-trash-offsets/v2/robot_position.html#position-relative-to-trash-containers)
- simulated snippets in 2.16 / `edge`

# Changelog

- new § on Labware and Deck Positions page
- updated description and example for `move_to()` to cover new
acceptable `location`s
- version added statements for `TrashBin`, `WasteChute`, and their
methods in API Reference

# Review requests

- double check code snippets. not only simulate, but do what i say they
do?
- confirm version added for the methods. it feels like we may need to
gate this to 2.17, because otherwise there will be a big functionality
gap between robot system 7.1 (which can't do any of this) and 7.2 (which
can). the classes are unambiguously _New in version 2.16._

# Risk assessment

low…but see API version gate caveat above.

ecormany committed:
chore: re-publish trash container top() methods

ecormany committed:
docs(api): update descriptions of partial tip pickup behavior for 7.3 (#15003)

Update partial tip pickup docs to reflect the latest behavior, as added
in #15009 and to be released in robot software 7.3.

Addresses RTC-440.

[Sandbox](http://sandbox.docs.opentrons.com/docs-partial-pickup-channel-1/v2/)

- Removed "Column 12" and "Column 1" headers from Partial Tip Pickup
page. Removed tip-tracking code for Column 1 as no longer needed..
- Reworded note in `configure_nozzle_layout` API reference entry.

Already spoke with @CaseyBatten to get an idea of what needed to be
included here and what will go in separate documentation.

- <s>Should we actually put this straight into `edge` and deploy ASAP,
or can it wait a month and roll out with 2.18 docs (current plan)?</s>
PAPI 2.18 deploy will be timed with robot stack 7.3 release, which this
aligns with.

none

ecormany committed:
docs(api): new behavior for `Labware.set_offset()` in PAPI 2.18 (#14967)

# Overview

Documents [updated behavior](#14940) for `Labware.set_offset()` in
Python API version 2.18.

Addresses RTC-234

# Test Plan

Sandbox:
- [API reference
entry](http://sandbox.docs.opentrons.com/docs-set_offset-updates/v2/new_protocol_api.html#opentrons.protocol_api.Labware.set_offset)
- [new
section](http://sandbox.docs.opentrons.com/docs-set_offset-updates/v2/new_advanced_running.html#labware-offset-behavior)
of Advanced Control
 
# Changelog

- adapted new docstring draft written by @sfoster1 
- adapt sample code and move it to Advanced Control

# Review requests

have we accurately captured the three phases of behavior (2.12–13,
14–17, 18+)?

# Risk assessment

none

ecormany committed:
docs(api): describe new `drop_tip()` behavior introduced in PAPI 2.18 (#15078)

# Overview

Describe the new `apiLevel`-gated behavior for `drop_tip()` in Python
API 2.18.

Addresses RTC-448

# Test Plan

[Sandbox API ref
entry](http://sandbox.docs.opentrons.com/docstring-drop-tip-alternating-2.18/v2/new_protocol_api.html#opentrons.protocol_api.InstrumentContext.drop_tip)

# Changelog

Move ¶ about change in 2.15 down, and add another ¶ about 2.18.

# Review requests

This is how it works, right?

# Risk assessment

None, docstring only.

ecormany committed:
docs(api): runtime parameters in Python API 2.18 (#14677)

# Overview

Full documentation for setting up and using runtime parameters in Python
protocols.

Closes RTC-416, RTC-428, RTC-429, RTC-430, RTC-451

# Test Plan

- Check content and formatting of all pages in the
[sandbox](http://sandbox.docs.opentrons.com/docs-rtp/v2/runtime_parameters.html)
- Simulate all [code from use case
pages](https://gist.github.com/ecormany/9f4d871851ec4baceb9fae177663dd00).

# Changelog

- New landing page for Runtime Parameters
- 6 new subpages
- Unhide `ProtocolContext.params` in API Reference. ~Note, this still
needs a docstring, either here or in a separate PR.~ Added
[docstring](http://sandbox.docs.opentrons.com/docs-rtp/v2/new_protocol_api.html#opentrons.protocol_api.ProtocolContext.params).

# Review requests

Read some or all of it, try the code, tell me what can be better.

# Risk assessment

None

---------

Co-authored-by: Sanniti Pimpley <sanni-t@users.noreply.github.com>

ecormany committed:
docs(api): Versioning page updates for Python API 2.18 (#15183)

# Overview

Summarize all of the changes in Python API 2.18 on the Versioning
documentation page.

Closes RTC-463

# Test Plan


[Sandbox](http://sandbox.docs.opentrons.com/docs-versioning-2.18/v2/versioning.html)

# Changelog

- add row to version table
- feature bullets

# Review requests

any glaring omissions?

# Risk assessment

nil

ecormany committed:
docs(api): explain order of operations for RTP value updates triggering reanalysis (#15195)

# Overview

Testing revealed that labware offsets aren't applied to labware that are
only loaded as the result of custom RTP values. This is somewhat
expected, due to the order of operations: analyze -> apply offsets ->
choose RTP values -> reanalyze.

This docs addition goes into much greater detail than the very short
summary we could fit into the release notes (see #15194).

# Test Plan


[Sandbox](http://sandbox.docs.opentrons.com/docs-rtp-plate-swap/v2/parameters/using_values.html#limitations)

# Review requests

Is this too much? Will it all be rendered moot by a hotfix?

# Risk assessment

none

jbleon95 reviewed (approved):
Did a quick scan through, everything looks accurate!

ecormany committed:
docs(api): fix type specification for `add_parameters()` (#15255)

# Overview

I got the type specification for the object of `add_parameters()` wrong.

# Test Plan

Simulated a protocol that uses the `def add_parameters(…` line verbatim
on 7.3.0-b1.

# Changelog

1-line diff

# Risk assessment

none

ecormany committed:
chore: bump API version to 2.18 in conf.py (#15261)

# Overview

This is a manual process now! Didn't forget this time. `|apiLevel|` will
now be replaced with `2.18`.

# Test Plan


[Sandbox](http://sandbox.docs.opentrons.com/docs-bump-version-218/v2/versioning.html#maximum-supported-versions)
<- mostly observe the text replacement in the max supported version
ranges.

# Changelog

- new API version
- new robot software version!

# Risk assessment

nil
